File: multiStructure/frameworks/ts-playwright/pages/checkout/checkoutInformationPage.ts
import { type Page, type Locator } from '@playwright/test';

export interface UserDetails {
  firstname: string;
  lastname: string;
  zipcode: string;
}

export class CheckoutInformation {
  readonly page: Page;
  readonly title: Locator;
  readonly firstName: Locator;
  readonly lastName: Locator;
  readonly zipcode: Locator;
  readonly continueButton: Locator;

  constructor(page: Page) {
    this.page = page;
    this.title = page.locator('.title');
    this.firstName = page.locator('#first-name');
    this.lastName = page.locator('#last-name');
    this.zipcode = page.locator('#postal-code');
    this.continueButton = page.locator('#continue');
  }

  async getTitleText(): Promise<string> {
    return (await this.title.first().textContent()) ?? '';
  }

  async enterDetails(data: UserDetails): Promise<void> {
    await this.firstName.fill(data.firstname);
    await this.lastName.fill(data.lastname);
    await this.zipcode.fill(data.zipcode);
  }

  async clickOnContinueButton(): Promise<void> {
    await this.continueButton.click();
  }
}

File: multiStructure/frameworks/csharp-playwright/Pages/CheckoutInformationPage.cs
using Microsoft.Playwright;
using PlaywrightFramework.Utils.DataFactory;

namespace PlaywrightFramework.Pages;

public class CheckoutInformationPage
{
    private readonly IPage _page;

    public CheckoutInformationPage(IPage page)
    {
        _page = page;
    }

    private ILocator Title => _page.Locator(".title").First;
    private ILocator FirstName => _page.Locator("#first-name");
    private ILocator LastName => _page.Locator("#last-name");
    private ILocator ZipCode => _page.Locator("#postal-code");
    private ILocator ContinueButton => _page.Locator("#continue");

    public async Task<string> GetTitleTextAsync()
    {
        return await Title.TextContentAsync() ?? string.Empty;
    }

    public async Task EnterDetailsAsync(UserData data)
    {
        await FirstName.FillAsync(data.FirstName);
        await LastName.FillAsync(data.LastName);
        await ZipCode.FillAsync(data.ZipCode);
    }

    public async Task ClickOnContinueButtonAsync()
    {
        await ContinueButton.ClickAsync();
    }
}